apps/frontend/src/pages/analytics/analytics.tsx

```diff
@@ -1,28 +1,72 @@
-import { DateInput, PageLayout } from "~/libs/components/components.js";
+import { DateInput, Loader, PageLayout } from "~/libs/components/components.js";
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
+import { DataStatus } from "~/libs/enums/enums.js";
 import { subtractDays } from "~/libs/helpers/helpers.js";
-import { useAppForm, useCallback } from "~/libs/hooks/hooks.js";
+import {
+	useAppDispatch,
+	useAppForm,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useState,
+} from "~/libs/hooks/hooks.js";
+import { actions as activityLogActions } from "~/modules/activity-logs/activity-logs.js";
 
+import { AnalyticsTable } from "./libs/components/components.js";
 import { ANALYTICS_DATE_MAX_RANGE } from "./libs/constants/constants.js";
 import styles from "./styles.module.css";
 
 const Analytics = (): JSX.Element => {
+	const dispatch = useAppDispatch();
 	const todayDate = new Date();
 
+	const initialDateRange: [Date, Date] = [
+		subtractDays(todayDate, ANALYTICS_DATE_MAX_RANGE),
+		todayDate,
+	];
+
+	const { activityLogs, dataStatus } = useAppSelector(
+		({ activityLogs }) => activityLogs,
+	);
+
+	const [dateRange, setDateRange] = useState<[Date, Date]>(initialDateRange);
+
 	const { control, handleSubmit } = useAppForm({
 		defaultValues: {
-			dateRange: [subtractDays(todayDate, ANALYTICS_DATE_MAX_RANGE), todayDate],
+			dateRange: initialDateRange,
 		},
 	});
 
+	const handleLoadActivityLogs = useCallback(() => {
+		void dispatch(activityLogActions.loadAll());
+	}, [dispatch]);
+
+	useEffect(() => {
+		handleLoadActivityLogs();
+	}, [handleLoadActivityLogs]);
+
+	const handlePopoverClose = useCallback(
```

Comment:
Agree with Vlad, we use useForm for submission, no need to create additional functions, data will be already present in values of form. There will also be other inputs, so use of this is more convenient.